Jenkinsfile

```diff
@@ -0,0 +1,131 @@
+@Library('xmos_jenkins_shared_library@v0.27.0')
+
+def runningOn(machine) {
+  println "Stage running on:"
+  println machine
+}
+
+getApproval()
+pipeline {
+  agent none
+
+  parameters {
+    string(
+      name: 'TOOLS_VERSION',
+      defaultValue: '15.2.1',
+      description: 'The XTC tools version'
+    )
+    booleanParam(
+      name: 'XMATH_SMOKE_TEST',
+      defaultValue: true,
+      description: 'Enable smoke run'
+    )
+  } // parameters
+  options {
+    skipDefaultCheckout()
+    timestamps()
+    buildDiscarder(xmosDiscardBuildSettings())
+  } // options
+
+  stages {
+    stage('Bullds and tests') {
+      parallel {
+        stage('Linux builds and tests') {
+          agent {
+            label 'xcore.ai'
+          }
+          stages {
+            stage('Build') {
+              steps {
+                runningOn(env.NODE_NAME)
+                dir('lib_xcore_math') {
+                  checkout scm
+                  // fetch submodules
+                  sh 'git submodule update --init --recursive --jobs 4'
+                  withTools(params.TOOLS_VERSION) {
+                    // xs3a build
+                    sh "cmake -B build_xs3a -DXMATH_SMOKE_TEST=${params.XMATH_SMOKE_TEST} --toolchain=etc/xmos_cmake_toolchain/xs3a.cmake"
+                    sh 'make -C build_xs3a -j4'
+                    // x86 build
+                    sh "cmake -B build_x86 -DXMATH_SMOKE_TEST=${params.XMATH_SMOKE_TEST}"
+                    sh 'make -C build_x86 -j4'
+                    // xmake build
+                    dir('test/legacy_build') {
+                      sh 'xmake -j4'
+                      sh 'xrun --io --id 0 bin/legacy_build.xe'
+                    }
+                  }
+                }
+              }
+            } // Build
+
+            stage('Unit tests xs3a') {
+              steps {
+                dir('lib_xcore_math/build_xs3a/test') {
+                  withTools(params.TOOLS_VERSION) {
+                    sh 'xrun --xscope --id 0 --args bfp_tests/bfp_tests.xe        -v'
```

Comment:
what does "-v" option do? Doesn't it just print the version? If you want to increase the verbosity, you need to use "--verbose", but I never used it myself,

Reply:
`-v` here is for the test itself, not `xrun`, hence it has `--args`. It means verbose